pkg/catalog/endpoint.go

```diff
@@ -38,33 +38,40 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	return endpoints, nil
 }
 
-// ListAllowedEndpointsForService returns only those endpoints for a service that belong to the allowed outbound service accounts
+// ListEndpointsForServiceIdentity returns only those endpoints for a allowed outbound service accounts
 // for the given downstream identity
 // Note: ServiceIdentity must be in the format "name.namespace" [https://github.com/openservicemesh/osm/issues/3188]
-func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+func (mc *MeshCatalog) ListEndpointsForServiceIdentity(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
 	outboundEndpoints, err := mc.listEndpointsForService(upstreamSvc)
 	if err != nil {
 		log.Error().Err(err).Msgf("Error looking up endpoints for upstream service %s", upstreamSvc)
 		return nil, err
 	}
+	outboundEndpointsSet := make(map[string]endpoint.Endpoint)
+	for _, ep := range outboundEndpoints {
+		outboundEndpointsSet[ep.String()] = ep
+	}
 
-	destSvcAccounts, err := mc.ListAllowedOutboundServiceIdentities(downstreamIdentity)
+	destSvcIdentities, err := mc.ListAllowedOutboundServiceIdentities(downstreamIdentity)
 	if err != nil {
 		log.Error().Err(err).Msgf("Error looking up outbound service accounts for downstream identity %s", downstreamIdentity)
 		return nil, err
 	}
 
 	// allowedEndpoints comprises of only those endpoints from outboundEndpoints that matches the endpoints from listEndpointsForServiceIdentity
 	// i.e. only those interseting endpoints are taken into cosideration
+	destPodEndpointsSet := make(map[string]endpoint.Endpoint)
```

Comment:
you are not using `destPodEndpointsSet` to really lookup anything (you end up walking it), so this map could just remain as a slice and avoid the N walk?

Reply:
Right. `destPodEndpointSet` is removed.